gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -2266,16 +2397,35 @@ bool GazeboRosApiPlugin::spawnAndConform(TiXmlDocument &gazebo_model_xml, std::s
   // todo: should wait for response response_sub_, check to see that if _msg->response == "nonexistant"
 
   gazebo::physics::ModelPtr model = world_->GetModel(model_name);
-  if (model)
+  gazebo::physics::LightPtr light = world_->Light(model_name);
+  if ((isLight && light != NULL) || (model != NULL))
   {
     ROS_ERROR_NAMED("api_plugin", "SpawnModel: Failure - model name %s already exist.",model_name.c_str());
     res.success = false;
-    res.status_message = "SpawnModel: Failure - model already exists.";
+    res.status_message = "SpawnModel: Failure - entity already exists.";
     return true;
   }
 
-  // Publish the factory message
-  factory_pub_->Publish(msg);
+  #if GAZEBO_MAJOR_VERSION > 6
```

Comment:
i think we shouldn't support versions older than 6 for the kinetic branch, see https://github.com/ros-simulation/gazebo_ros_pkgs/pull/542